src/nv_ingest/stages/nim/infographic_extraction.py

```diff
@@ -0,0 +1,254 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import functools
+import logging
+import traceback
+from concurrent.futures import ThreadPoolExecutor
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+import pandas as pd
+from morpheus.config import Config
+
+from nv_ingest.schemas.infographic_extractor_schema import InfographicExtractorSchema
+from nv_ingest.stages.multiprocessing_stage import MultiProcessingBaseStage
+from nv_ingest.util.image_processing.transforms import base64_to_numpy
+from nv_ingest.util.nim.helpers import NimClient
+from nv_ingest.util.nim.helpers import create_inference_client
+from nv_ingest.util.nim.paddle import PaddleOCRModelInterface
+
+logger = logging.getLogger(__name__)
+
+PADDLE_MIN_WIDTH = 32
+PADDLE_MIN_HEIGHT = 32
+
+
+def _update_metadata(
+    base64_images: List[str],
+    paddle_client: NimClient,
+    worker_pool_size: int = 8,  # Not currently used
+    trace_info: Dict = None,
+) -> List[Tuple[str, Tuple[Any, Any]]]:
+    """
+    Given a list of base64-encoded images, this function filters out images that do not meet the minimum
+    size requirements and then calls the PaddleOCR model via paddle_client.infer to extract infographic data.
+
+    For each base64-encoded image, the result is:
+        (base64_image, (text_predictions, bounding_boxes))
+
+    Images that do not meet the minimum size are skipped (resulting in ("", "") for that image).
+    The paddle_client is expected to handle any necessary batching and concurrency.
+    """
+    logger.debug(f"Running infographic extraction using protocol {paddle_client.protocol}")
+
+    # Initialize the results list in the same order as base64_images.
+    results: List[Optional[Tuple[str, Tuple[Any, Any, Any]]]] = [("", None, None)] * len(base64_images)
+
+    valid_images: List[str] = []
+    valid_indices: List[int] = []
+
+    # Pre-decode image dimensions and filter valid images.
+    for i, img in enumerate(base64_images):
+        array = base64_to_numpy(img)
+        height, width = array.shape[0], array.shape[1]
+        if width >= PADDLE_MIN_WIDTH and height >= PADDLE_MIN_HEIGHT:
+            valid_images.append(img)
+            valid_indices.append(i)
+        else:
+            # Image is too small; mark as skipped.
+            results[i] = (img, None, None)
+
+    # Prepare data payloads for both clients.
+    data_paddle = {"base64_images": valid_images}
+
+    _ = worker_pool_size
+    with ThreadPoolExecutor(max_workers=2) as executor:
```

Comment:
Same question about paddle.

Also, we don't need a ThreadPoolExecutor here. The only reason we use it in places like the chart extractor is so we can run two models in parallel.

Reply:
Removed in [8bc3c7d](https://github.com/NVIDIA/nv-ingest/pull/483/commits/8bc3c7d5263d659a580db16073357115b22db0b1)